File: sdks/go/tests/conformance/identity_domain_helpers_test.go
package conformance

import (
	"testing"

	"github.com/astraive/loza/sdks/go"
)

func TestIdentityDomainHelpers(t *testing.T) {
	attrCases := []struct {
		name string
		attr loza.Attr
		key  string
	}{
		{"user id", loza.UserID("u_123"), "user.id"},
		{"tenant id", loza.TenantID("t_123"), "tenant.id"},
		{"session id", loza.SessionID("s_123"), "session.id"},
		{"request id", loza.RequestID("req_123"), "request_id"},
		{"trace id", loza.TraceID("trace_123"), "trace_id"},
		{"span id", loza.SpanID("span_123"), "span_id"},
		{"order id", loza.OrderID("ord_123"), "order.id"},
		{"cart id", loza.CartID("cart_123"), "cart.id"},
		{"payment id", loza.PaymentID("pay_123"), "payment.id"},
		{"subscription id", loza.SubscriptionID("sub_123"), "subscription.id"},
		{"invoice id", loza.InvoiceID("inv_123"), "invoice.id"},
		{"job id", loza.JobID("job_123"), "job.id"},
		{"message id", loza.MessageID("msg_123"), "message.id"},
		{"correlation id", loza.CorrelationID("corr_123"), "correlation.id"},
		{"deployment id", loza.DeploymentID("dep_123"), "deployment_id"},
		{"release", loza.Release("1.2.3"), "release"},
		{"region", loza.Region("us-east-1"), "region"},
		{"checkout cart item count", loza.CheckoutCartItemCount(3), "checkout.cart_item_count"},
		{"payment method", loza.PaymentMethod("card"), "payment.method"},
		{"billing plan", loza.BillingPlan("pro"), "billing.plan"},
		{"agent model", loza.AgentModel("gpt-5.5"), "agent.model"},
		{"rag index", loza.RAGIndex("docs"), "rag.index"},
	}

	for _, tc := range attrCases {
		t.Run(tc.name, func(t *testing.T) {
			if tc.attr.Key != tc.key {
				t.Fatalf("expected key %q, got %q", tc.key, tc.attr.Key)
			}
		})
	}
}

File: sdks/js/tests/identity_domain_helpers.test.ts
import { afterEach, it } from 'node:test';
import assert from 'node:assert/strict';
import * as loza from '../src/index.ts';

afterEach(() => {
  loza.reset();
});

it('covers identity and domain helpers', () => {
  assert.equal(loza.userId('u_123').key, 'user.id');
  assert.equal(loza.tenantId('t_123').key, 'tenant.id');
  assert.equal(loza.sessionId('s_123').key, 'session.id');
  assert.equal(loza.requestId('req_123').key, 'request_id');
  assert.equal(loza.traceId('trace_123').key, 'trace_id');
  assert.equal(loza.spanId('span_123').key, 'span_id');
  assert.equal(loza.orderId('ord_123').key, 'order.id');
  assert.equal(loza.cartId('cart_123').key, 'cart.id');
  assert.equal(loza.paymentId('pay_123').key, 'payment.id');
  assert.equal(loza.subscriptionId('sub_123').key, 'subscription.id');
  assert.equal(loza.invoiceId('inv_123').key, 'invoice.id');
  assert.equal(loza.jobId('job_123').key, 'job.id');
  assert.equal(loza.messageId('msg_123').key, 'message.id');
  assert.equal(loza.correlationId('corr_123').key, 'correlation.id');
  assert.equal(loza.commitSha('abc123').key, 'deployment.commit_sha');
  assert.equal(loza.release('1.2.3').key, 'release');
  assert.equal(loza.checkoutCartItemCount(3).key, 'checkout.cart_item_count');
  assert.equal(loza.paymentProvider('stripe').key, 'payment.provider');
  assert.equal(loza.billingPlan('pro').key, 'billing.plan');
  assert.equal(loza.agentModel('gpt-5.5').key, 'agent.model');
  assert.equal(loza.ragIndex('docs').key, 'rag.index');
});
